doc/test-framework/writing-e2e-tests.md

```diff
@@ -0,0 +1,236 @@
+# Using the Operator SDK's Test Framework to Write E2E Tests
+
+End-to-end tests are an essential tool to ensure that an operator works
+as intended in real-world scenarios. The Operator SDK includes a testing
+framework to make writing tests simpler and quicker by removing boilerplate
+code and providing common test utilities. The Operator SDK includes the
+test framework as a library under `pkg/test` and the e2e tests are written
+as standard go tests.
+
+## Components
+
+The test framework includes a few components. The most important to talk
+about are Framework and TestCtx.
+
+### Framework
+
+[Framework][framework-link] contains all global variables, such as the kubeconfig, kubeclient,
+scheme, and dynamic client (provided via the controller-runtime project).
+It is initialized by MainEntry and can be used anywhere in the tests.
+
+### TestCtx
+
+[TestCtx][testctx-link] is a local context that stores important information for each test, such
+as the namespace for that test and the finalizer (cleanup) functions. By handling
+namespace and resource initialization through TestCtx, we can make sure that all
+resources are properly handled and removed after the test finishes.
+
+## Walkthrough: Writing Tests
+
+In this section, we will be walking through writing the e2e tests of the sample
+[memcached-operator][memcached-sample].
+
+### Main Test
+
+The first step to writing a test is to create the `main_test.go` file. The `main_test.go`
+file simply calls the test framework's main entry that sets up the framework and then
+starts the tests. It should be pretty much identical for all operators. This is what it
+looks like for the memcached-operator:
+
+```go
+package e2e
+
+import (
+    "testing"
+
+    f "github.com/operator-framework/operator-sdk/pkg/test"
+)
+
+func TestMain(m *testing.M) {
+    f.MainEntry(m)
+}
+```
+
+### Individual Tests
+
+In this section, we will be designing a test based on the [memcached_test.go][memcached-test-link] file
+from the [memcached-operator][memcached-sample] sample.
+
+#### 1. Import the framework
+
+Once MainEntry sets up the framework, it runs the remainder of the tests. First, make
+sure to import `testing`, the operator-sdk test framework (`pkg/test`) as well as your operator's libraries:
+
+```go
+import (
+    "testing"
+
+    cachev1alpha1 "github.com/operator-framework/operator-sdk-samples/memcached-operator/pkg/apis/cache/v1alpha1"
+
+    framework "github.com/operator-framework/operator-sdk/pkg/test"
+)
+```
+
+#### 2. Register types with framework scheme
+
+The next step is to register your operator's scheme with the framework's dynamic client.
+To do this, you need to create a list struct for your custom resource and pass the list
+object and the object's addToScheme function to the framework's [AddToFrameworkScheme][scheme-link]
```

Comment:
Rephrase 
"you need to create a list struct for your custom resource and pass the list
object and the object's addToScheme function to the framework's ..."
to:
"pass the CRD's `AddToScheme` function and its List type object to the framework's ..."